webdocs/introduction/why/_index.md

```diff
@@ -0,0 +1,32 @@
+---
+title: Why the plugin exists
+github_repo: https://github.com/ctfer-io/ctfd-chall-manager
+github_project_repo: https://github.com/ctfer-io/ctfd-chall-manager
+description: >
+  A quick introduction in to what CTFd-chall-manager is and why it should be used.
+weight: 1
+resources:
+  - src: "**.png"
+  - src: "**.gif"
+categories: [How-to Guides]
+---
+
+## Why Chall-Manager exists ?
+
+Originally, [CTFer.io](https://ctfer.io) used Pulumi in Go to deploy CTFd in a Kubernetes cluster due to our software development expertise. One of Chall-Manager's target capabilities is "anything deployable by Pulumi can be managed by Chall-Manager." This promise would have been difficult to achieve if we used only Python, as it requires many dependencies and would force each plugin user to rebuild the CTFd Docker image.
+
+Additionally, we aim for an Offline-First model, and using Golang makes it easier to achieve this goal because we can compile the program and deploy it in a Docker image.
+
+However, executing Pulumi with Go inside CTFd was not possible, so we had to create a dedicated system.
```

Comment:
It was actually possible. The real reason was building a reusable component, non-vendor lock-in, and without hard coupling between technical objects with different goals.